### PR TITLE
add main functionality todo app

### PR DESCRIPTION
**Pull request by AnastasiiaYaitska**

### CODE REVIEW

File: react/react1/week2/toDoApp/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const formSubmit = (todo) => {
+    if (!todo.trim()) {
+      return;
+    }
+    const newTodo = {
+      id: nanoid(),
+      description: todo,
+    };
+    setTodoList((prevState) => {
+      return [newTodo, ...prevState];
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/App.jsx:22]: Good use of the dot operator

File: react/react1/week2/toDoApp/src/components/AddRandomTask/AddRandomTask.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const AddRandomTask = ({ addTask }) => {
+  return (
+    <button type="button" onClick={addTask}>
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/AddRandomTask/AddRandomTask.jsx:5]: Good to see you use a function as a prop 👍

File: react/react1/week2/toDoApp/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  //     return [
+  //       {
+  //         id: nanoid(),
+  //         description: "Random task",
+  //       },
+  //       ...prevState,
+  //     ];
+  //   });
+  // };
+  //If you uncomment all comments you will see extra button for adding random task. It is functionality from exercise , but I add some form. Hope it is ok
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/App.jsx:37]: Perfectly fine 👍

File: react/react1/week2/toDoApp/src/components/AddTodoForm/AddTodoForm.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+
+const AddTodoForm = ({ formSubmit }) => {
+  const [newTask, setNewTask] = useState("");
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/AddTodoForm/AddTodoForm.jsx:4]: good use of state

File: react/react1/week2/toDoApp/src/components/Header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from "react";
+
+const Header = ({ title }) => {
+  const [seconds, setSeconds] = useState(0);
+
+  useEffect(() => {
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/Header/Header.jsx:6]: Great solution to the timer 🥇 Good to see you use a return in your useEffect for cleanup

File: react/react1/week2/toDoApp/src/components/TodoList/TodoList.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import Todo from "../Todo/Todo";
+
+const TodoList = ({ todos, deleteTask }) => {
+  return (
+    <ul>
+      {todos.length ? (
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/TodoList/TodoList.jsx:7]: Good use of inline ternary operator 👍

File: react/react1/week2/toDoApp/src/components/Todo/Todo.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from "react";
+import { Label } from "./Todo.styled";
+
+const Todo = ({ todo, deleteTask }) => {
+  const [isCheck, setIsCheck] = useState(false);
+  const { id, description } = todo;
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/Todo/Todo.jsx:6]: Good use of destructuring

File: react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import styled from "styled-components";
+
+export const Label = styled.label`
```
Comment (LucasIversen) [inline react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js:3]: There are other ways to make our style adaptive that might be a bit easier. I'd have used inline style in this instance

**Review by LucasIversen — APPROVED** [review]
Hi Anastasiia.

Really good job! You seem to have really caught on to react fast, and show a good use of all the concepts we've presented in class! Keep it up 💪

File: react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js (hunk 8 above)
Comment (AnastasiiaYaitska) [reply react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js:3]: I thought about inline styles already after the pull request.🙈 Should I change it to inline? Or can I merge the branch?

Comment (LucasIversen) [reply react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js:3]: This also works, so it's up to you :)

Comment (AnastasiiaYaitska) [reply react/react1/week2/toDoApp/src/components/Todo/Todo.styled.js:3]: Ok. Thank you!

**Pull request merged**